store/boltdb/boltdb.go

```diff
@@ -302,6 +301,8 @@ func (b *BoltDB) List(keyPrefix string) ([]*store.KVPair, error) {
 	})
 	if len(kv) == 0 {
 		return nil, store.ErrKeyNotFound
+	} else if len(kv) == 1 && kv[0].Key == keyPrefix {
+		return []*store.KVPair{}, err
```

Comment:
Unfortunately this is not going to work here: if we have an array of > 1 element, we're still going return the directory key like before. I think we should do like other stores and filter in the loop above before appending to the array of pairs, when `key == keyPrefix` we skip that entry and continue looping over keys.